### PR TITLE
(FACT-2726) Add solaris dmi facts

### PR DESCRIPTION
**Pull request by florindragos**

### CODE REVIEW

**Comment by florindragos** [comment]
Jenkins please test this on all

File: lib/facter/facts/solaris/dmi/product/name.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          FACT_NAME = 'dmi.product.name'
+          ALIASES = 'productname'
+
+          def call_the_resolver
+            fact_value = nil
+            if isa == 'i386'
+              fact_value = Facter::Resolvers::Solaris::Dmi.resolve(:product_name)
+            elsif isa == 'sparc'
+              fact_value = Facter::Resolvers::Solaris::DmiSparc.resolve(:product_name)
+            end
```
Comment (oanatmaria) [inline lib/facter/facts/solaris/dmi/product/name.rb:17]: you can do something like:
`fact_value = if isa == 'i386'
                 Facter::Resolvers::Solaris::Dmi.resolve(:product_name)
            elsif isa == 'sparc'
                Facter::Resolvers::Solaris::DmiSparc.resolve(:product_name)
            end`

File: lib/facter/resolvers/solaris/dmi_resolver.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Facter
+  module Resolvers
+    module Solaris
+      class Dmi < BaseResolver
```
Comment (IrimieBogdan) [inline lib/facter/resolvers/solaris/dmi_resolver.rb:6]: We should drop `_resolver` from file name.

File: lib/facter/resolvers/solaris/dmi_sparc_resolver.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Facter
+  module Resolvers
+    module Solaris
+      class DmiSparc < BaseResolver
```
Comment (IrimieBogdan) [inline lib/facter/resolvers/solaris/dmi_sparc_resolver.rb:6]: We should drop `resolver` from file name.

File: lib/facter/resolvers/solaris/dmi_sparc.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            output = exec_prtdiag
+            return unless output
+
+            matches = output.match(/System Configuration:\s+(.+?)\s+sun\d+\S+\s+(.+)/)&.captures
+
+            @fact_list[:manufacturer] = matches[0]&.strip
+            @fact_list[:product_name] = matches[1]&.strip
+
+            sneep = exec_sneep&.strip
+            @fact_list[:serial_number] = sneep if sneep
```
Comment (IrimieBogdan) [inline lib/facter/resolvers/solaris/dmi_sparc.rb:27]: We should remove `if sneep` and attribute nil as value if the command cannot run. If we do this, `/usr/sbin/sneep` will only be called once even if we call the resolver multiple times asking for `serial_number`

**Comment by florindragos** [comment]
Jenkins please test this on all

**Comment by florindragos** [comment]
Jenkins please test this on all

File: lib/facter/facts/solaris/dmi/product/uuid.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+module Facts
+  module Solaris
+    module Dmi
+      module Product
+        class Uuid
+          FACT_NAME = 'dmi.product.uuid'
+          ALIASES = 'uuid'
+
+          def call_the_resolver
+            fact_value = nil
```
Comment (IrimieBogdan) [inline lib/facter/facts/solaris/dmi/product/uuid.rb:12]: This is not required

**Pull request merged**